concordium-sdk/src/main/java/com/concordium/sdk/responses/transactionstatus/RejectReasonAmountTooLarge.java

```diff
@@ -0,0 +1,39 @@
+package com.concordium.sdk.responses.transactionstatus;
+
+import com.concordium.sdk.transactions.GTUAmount;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Getter;
+import lombok.ToString;
+
+import java.util.List;
+import java.util.Map;
+
+@Getter
+@ToString
+public class RejectReasonAmountTooLarge extends RejectReason {
+
+    private AbstractAccount account;
+    private GTUAmount amount;
+
+    @JsonCreator
+    RejectReasonAmountTooLarge(@JsonProperty("contents") List<Object> contents) {
+        if (contents.size() != 2) {
+            throw new IllegalArgumentException("Cannot parse AmountTooLarge. Unexpected array length.");
+        }
+        for (Object content : contents) {
+            if (content instanceof String) {
+                this.amount = GTUAmount.fromMicro((String) content);
+            } else if (content instanceof Map) {
+                this.account = AbstractAccount.parseAccount((Map<String, Object>) content);
+            } else {
+                throw new IllegalArgumentException("Cannot parse AmountTooLarge. Unexpected type.");
+            }
+        }
```

Comment:
Couldn't you just get the first and second element of the array here and cast it to the appropriate types?

It seems needlessly complex to iterate over the array (when you've already checked it has exactly 2 elements) and then have all the cases all the time in the loop.

Reply:
Yes, that will be cleaner.